### DISCUSSION TITLE
Tailwind syntax not supported by PostCSS Language Support in VSCode

### DISCUSSION THREAD

**Discussion by paulkre** [tailwindlabs/tailwindcss#6427] [Help] [answered]
[The docs say](https://tailwindcss.com/docs/editor-setup#syntax-support) that syntax support in VSCode can be enabled with the [PostCSS Language Support](https://marketplace.visualstudio.com/items?itemName=csstools.postcss)  plugin. But in my setup the plugin marks all the Tailwind CSS rules as errors. Is this just an issue with my setup or does the plugin really not support Tailwind?

<img width="204" alt="Screenshot 2021-12-11 at 19 00 54" src="https://user-images.githubusercontent.com/13008492/145686876-1d2615ed-ed21-4ddb-a5bd-426d11bf5053.png">

**Comment by adamwathan** [comment]
Make sure you change the default language for CSS files to PostCSS and that stuff should go away. If not you may also have to disable the built in CSS linting in VS Code.

Going to close as not an issue in the docs per se, better suited as a discussion in our main discussion area I think ❤️

**Comment by paulkre** [comment]
I'm pretty sure the errors come directly from the extension. If I disable it those errors become warnings. VSCode definitely regards the file as PostCSS. I'm just saying it might be a good idea to remove the recommendation for the extension from the docs because it either doesn't work or is complicated to set up. The sentence "VS Code has a PostCSS Language Support plugin you can use that works great with Tailwind CSS" is simply not correct.

**Comment by adamwathan** [comment]
This is what I see when the language is set to CSS:

![image](https://user-images.githubusercontent.com/4323180/145722939-9bab262f-93a9-4020-a63a-52ef59c51b90.png)

And this is what I see when it is set to PostCSS:

![image](https://user-images.githubusercontent.com/4323180/145722949-f28e5b8d-e473-4e3e-ab26-9d8df7f948fb.png)

This is with no other CSS options edited, so not sure what the difference would be.

**Comment by paulkre** [comment, marked as answer]
Oh wow, there was another extension ([postcss-sugarss-language](https://marketplace.visualstudio.com/items?itemName=mhmadhamster.postcss-language)) enabled. As soon as I disabled it, the errors went away. Sorry for the confusion and thanks for your help.

**Comment by paulkre** [comment]
Only now there is no autocomplete for regular CSS attributes. 😕  
I wonder, if there is a perfect VS Code setup that supports the development for both Tailwind and regular CSS.
